### DISCUSSION TITLE
Publish WPF or UWP to single .exe?

### DISCUSSION THREAD

**Discussion by TonyK-FmGroup** [unoplatform/uno#8595] [Q&A]
Hi,
We have a need to provide with the WPF or UWP app as a single .exe for windows?
I've tried following instructions from Microsoft, and others, on this topic but I always end up with multiple files.

**Comment by jeromelaban** [comment]
Thanks. Which instructions did you follow, specifically ?

**Reply by TonyK-FmGroup** [reply]
I've tried a few, also tried combining aspects on each. 

https://docs.microsoft.com/en-us/visualstudio/deployment/quickstart-deploy-using-clickonce-folder?view=vs-2022

https://docs.microsoft.com/en-us/dotnet/core/deploying/single-file/overview

https://stackoverflow.com/questions/69940627/deploy-with-vs2022-single-exe-file-for-net-6-wpf-application-not-works

It's Friday afternoon, so a good chance I've just done something wrong!

**Reply by jeromelaban** [reply]
Got it, that's a good start! Did you try building the app using the command line? Something like this:
```
dotnet publish -r win-x64 -p:PublishSingleFile=true --self-contained false
```

**Reply by TonyK-FmGroup** [reply]
That also created a lot of files.  I've also noticed messages like " warning N
U1701: Package 'SkiaSharp.Views.WPF 2.80.3' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFrame
work,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project tar
get framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project."

I created a .net 6 Uno solution and in the General Tab of the project properties for the WPF host shows the target framework as .net 6.0

**Reply by jeromelaban** [reply]
Thanks. Are you able to create a normal WPF app with a single file options, and does it actually create a single file ?

**Reply by TonyK-FmGroup** [reply]
That's a very good point!  I'll try that now and get back to you ASAP

**Reply by TonyK-FmGroup** [reply]
So far yes, although I don't have any dependencies so I'll try something a bit bigger. In the meantime, with the uno project, it seems to be the skia libraries that are the additional files. Plus assets/fonts/uno-fluentui-assets.  Is this potentially related to the above warning message?

**Reply by jeromelaban** [reply]
I see. Then in this case, there's something to fix in Uno to get there, as we don't have that embedded support yet.

**Reply by TonyK-FmGroup** [reply]
I see, thank you for the help. Will you raise an issue for this?

**Reply by jeromelaban** [reply]
We already do, in a larger epic: https://github.com/unoplatform/uno/issues/4309
